### ISSUE TITLE
Migration guide for v1-alpha.4 to v1-alpha.5 release

### ISSUE THREAD

**Issue by prydonius** [kubeapps/kubeapps#397]
Several breaking changes are going to be in this next release that we need to communicate well. The following changes should be highlighted:

- Helm Chart
  - need to explain reasoning for this change, and explain how to configure the chart to deploy a secure Tiller like the `kubeapps` CLI did
  - explain default option of pointing to the Helm CLI initialised Tiller for the users wanting this feature
- Switch from Helm CRD to Tiller Proxy
  - should not be too much of an issue for upgrades, but need to ensure people can use their existing release from the `kubeapps` CLI Tiller
- For Kubeapps deployed Tiller, switch from ConfigMap storage to Secrets and how to run migration script

I suggest we communicate this in the following ways:
- Migration guide added to docs
- Migration guide pointed to from GitHub release page and README
- (maybe) blog post promoting and explaining these changes

**Comment by prydonius** [comment]
@arapulido @andresmgot please add anything I might have missed

**Comment by andresmgot** [comment]
As far as I understood the Helm Chart is not a breaking change since we are going to maintain `kubeapps up` for at least the next version, isn't it? What I mean is that I don't think we need to add that to the migration guide, just explain the new way of deploying kubeapps in the main readme. 

Apart from that I would use a `CHANGELOG.md` for the migration guide that is kind of a more standard way of doing that. We can add breaking changes there.

**Comment by arapulido** [comment]
I agree with @andresmgot , we should keep both the chart and `kubeapps up` at least for a release. But yes, it should be well announced in the release notes

**Comment by prydonius** [comment]
It will be deprecated in the next release, but still supported for a couple more releases. With it being deprecated, we still should describe how to migrate to it because the idea is to get people to move to it and stop relying on the CLI. We can mention it's not required and the CLI can still be used, however.

**Comment by prydonius** [comment]
> Apart from that I would use a CHANGELOG.md for the migration guide that is kind of a more standard way of doing that. We can add breaking changes there.

A changelog is useful for summarising changes (breaking or not), but describing the steps to upgrade to a particular version is usually in a separate document (at least I haven't seen them in the same document before). We currently use GitHub releases for the changelog, I'd be open to moving that to a doc, but it's a separate issue.

**Comment by andresmgot** [comment]
> we still should describe how to migrate to it because the idea is to get people to move to it

You mean that somehow we need to make `helm install <repo>/kubeapps` work without failing because the elements of the chart already exists, isn't it? Is that something doable?

> A changelog is useful for summarising changes (breaking or not), but describing the steps to upgrade to a particular version is usually in a separate document (at least I haven't seen them in the same document before)

Yes, I suggested a changelog to have a file that doesn't look weird (and it's easy to find) after a few releases. If you think a different location is better I am fine with that as well.

**Comment by prydonius** [comment]
> You mean that somehow we need to make helm install <repo>/kubeapps work without failing because the elements of the chart already exists, isn't it? Is that something doable?

Good point, but not necessarily, it may be enough for an upgrade to be `kubeapps down`, `helm install ...` with the settings to enable Tiller to point to the `kubeapps` namespace so previous Tiller ConfigMaps are picked up. Something we need to look into. Of course, if the only way is to have a completely fresh installation then the sooner we communicate that the better.
